File: 08-Colas_de_prioridad_acotada/java/src/QueueNode.java
public class QueueNode<T> {
    private Node<T> front;
    private Node<T> back;
    private int length;

    public QueueNode() {
        front = back = null;
        length = 0;
    }

    public boolean isEmpty() {
        return front == null && length == 0;
    }

    public int length() {
        return length;
    }

    public T dequeue() {
        if(isEmpty()) {
            return null;
        }

        if(front == back) {
            back = null;
        }

        T topValue = front.getData();
        front = front.getNext();
        length--;
        return topValue;
    }

    public T front() {
        if(isEmpty()) {
            return null;
        }

        return front.getData();
    }

    public void enqueue(T value) {
        Node<T> newNode = new Node<T>(value);
        if(isEmpty()) {
            front = back = newNode;
        } else {
            back.setNext(newNode);
            back = newNode;
        }
        length++;
    }

    @Override
    public String toString() {
        String stackString = "";
        Node<T> iteratorNode = front;
        while(iteratorNode != null){
            stackString += iteratorNode;
            iteratorNode = iteratorNode.getNext();
        }
        return stackString;
    }
}

File: 08-Colas_de_prioridad_acotada/python/queueNode.py
from node import Node

class QueueNode:
    fontElement = None
    back = None
    length = 0

    def __init__(self):
        self.length = 0
        self.fontElement = None
        self.back = None

    def isEmpty(self):
        return self.fontElement == None and self.length == 0

    def getLength(self):
        return self.length

    def enqueue(self, valor):
        newNode = Node(valor)
        if self.isEmpty():
            self.fontElement = newNode
            self.back = newNode
        else:
            self.back.setNext(newNode)
            self.back = newNode
        self.length += 1

    def dequeue(self):
        if self.isEmpty():
            return

        if self.fontElement == self.back:
            self.back = None

        temp = self.fontElement.getData()
        self.fontElement = self.fontElement.getNext()
        self.length -= 1
        return temp

    def frontValue(self):
        if self.isEmpty():
            return
        return self.fontElement.getData()

    def __str__(self):
        stackString = "";
        iteratorNode = self.frontElement;
        while iteratorNode != None :
            stackString += iteratorNode;
            iteratorNode = iteratorNode.getNext();
        return stackString;
